### PR TITLE
restore timestamps in delta compression

### LINKED ISSUE #235: receivingvideo10s is broken
https://github.com/opentok/rtcstats-server/blob/develop/features.js#L787-L808
-- this assumes a getStats frequency of once per second. Should instead measure the time.

### PR DESCRIPTION
**Pull request by fippo**
also fixes #235 (the receiving*10s feature)

### CODE REVIEW

**Comment by fippo** [comment]
I am not sure if deleting the timestamps on the client side is a good idea. ISTR that stats for ended tracks would be kept around but the timestamp would never increase.

**Comment by fippo** [comment]
the code can now deal with clients that do not remove the timestamp.

File: features.js [hunk 1]
```diff
@@ -800,21 +800,27 @@ module.exports = {
...
-                        if (report.type === 'ssrc' && (report.kind === 'video' || report.mediaType === 'video') && id.indexOf('_recv')) {
-                            receivedVideo = {
-                                packetsReceived: report.packetsReceived,
-                                frameWidth: report.frameWidth,
-                                frameHeight: report.frameHeight
-                            };
-                        }
-                    });
+            if (peerConnectionLog[i].type === 'oniceconnectionstatechange' && !timeConnected) {
+                if (peerConnectionLog[i].value === 'connected' || peerConnectionLog[i].value === 'completed') {
```
Comment (ggarber) [inline features.js]: I find this approach more readable
```
const iceState = peerConnectionLog[i].value;
if (['connected', 'completed'].includes(iceState)]) {
```

File: features.js [hunk 2]
```diff
@@ -851,20 +857,26 @@ module.exports = {
...
-                    Object.keys(peerConnectionLog[i].value).forEach(id => {
-                        var report = peerConnectionLog[i].value[id];
-                        if (report.type === 'ssrc' && (report.kind === 'audio' || report.mediaType === 'audio') && id.indexOf('_recv')) {
-                            receivedAudio = {
-                                packetsReceived: report.packetsReceived,
-                                jitterbufferms: report.googJitterBufferMs
-                            };
-                        }
-                    });
+            if (peerConnectionLog[i].type === 'oniceconnectionstatechange' && !timeConnected) {
```
Comment (ggarber) [inline features.js]: These lines look very similar to th eones for video, a refactor could be a good idea while you change it

File: features.js (hunk 1 above)
Comment (fippo) [reply features.js]: that is one of the things enabled by es6. But remember this code was written when ES6 was not yet en vogue.

I'd rather avoid introducing too much es6 now and rather do a single bit "port it to es6" some day.

File: features.js (hunk 2 above)
Comment (fippo) [reply features.js]: this is copy-paste with different variable names for the purpose of readability...

**Pull request merged**